### PR TITLE
Add tests for the CrossCorrelate algorithm

### PR DESCRIPTION
**Pull request by gemmaguest**
This adds missing unit tests for the CrossCorrelate algorithm.

**To test:**

Run AlgorithmsTest_CrossCorrelateTest

Refs #18215 .

*Does not need to be in the release notes.*

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####

- [x] Is the code of an acceptable quality?
- [x] Does the code conform to the [coding standards](http://www.mantidproject.org/Coding_Standards)? Is it well structured with small focussed classes/methods/functions?
- [x] Are there unit/system tests in place? Are the unit tests small and test the a class in isolation?
- [ ] If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- [ ] Do changes function as described? Add comments below that describe the tests performed?
- [ ] How do the changes handle unexpected situations, e.g. bad input?
- [ ] Has the relevant documentation been added/updated?
- [ ] Is user-facing documentation written in a user-friendly manner?
- [ ] Has developer documentation been updated if required?

- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: Framework/Algorithms/test/CrossCorrelateTest.h [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+  // This tests an input X length of 3, which is the minimum the algorithm can
+  // handle
+  void testMinimumInputXLength() {
+    // setup and run the algorithm (includes basic checks)
+    CrossCorrelate alg;
+    const MatrixWorkspace_const_sptr inWS = setupAlgorithm(alg, 2.0, 3.5);
+    const MatrixWorkspace_const_sptr outWS = runAlgorithm(alg, inWS);
+
+    // specific checks
+    const Mantid::MantidVec outX = outWS->readX(0);
```
Comment (AntonPiccardoSelg) [inline Framework/Algorithms/test/CrossCorrelateTest.h]: You could consider using `const auto` here and in other places

File: Framework/Algorithms/test/CrossCorrelateTest.h [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+  }
+
+  void testInputXLength2() {
+    // this throws because at least 3 X values are required
+    CrossCorrelate alg;
+    setupAlgorithm(alg, 2.0, 3.0);
+    runAlgorithmThrows(alg);
+  }
+
+  void testInputXLength1() {
```
Comment (AntonPiccardoSelg) [inline Framework/Algorithms/test/CrossCorrelateTest.h]: Isn't this the same test as `testInputXLength2`?

**Comment by AntonPiccardoSelg** [comment]
Very nice test, just a couple of minor comments

File: Framework/Algorithms/test/CrossCorrelateTest.h (hunk 2 above)
Comment (gemmaguest) [inline Framework/Algorithms/test/CrossCorrelateTest.h]: It executes a different code path. I can add a comment to make that clear.

Comment (AntonPiccardoSelg) [inline Framework/Algorithms/test/CrossCorrelateTest.h]: No, that is not necessary then

File: Framework/Algorithms/test/CrossCorrelateTest.h (hunk 1 above)
Comment (gemmaguest) [inline Framework/Algorithms/test/CrossCorrelateTest.h]: Ah, I'm `using namespace Mantid` anyway, so I can just use `const MantidVec`. I would prefer that, unless there is a convention to use `auto`?

Comment (AntonPiccardoSelg) [inline Framework/Algorithms/test/CrossCorrelateTest.h]: The Mantid code that has been written more recently normally uses `auto`.

**Comment by AntonPiccardoSelg** [comment]
Code changes look good to me, the test are good to undestand. Please :shipit:

**Pull request merged**